crates/sui-storage/src/event_store/mod.rs

```diff
@@ -325,85 +326,95 @@ pub trait EventStore {
     /// which have sequence numbers below the current one will be skipped.  This feature
     /// is intended for deduplication.
     /// Returns Ok(rows_affected).
-    async fn add_events(&self, events: &[EventEnvelope]) -> Result<u64, SuiError>;
+    async fn add_events(&self, events: &mut [EventEnvelope]) -> Result<u64, SuiError>;
 
+    async fn all_events(
+        &self,
+        cursor: EventID,
+        limit: usize,
+        reverse: bool,
+    ) -> Result<Vec<StoredEvent>, SuiError>;
     /// Returns at most `limit` events emitted by a given
     /// transaction, sorted in order emitted.
     async fn events_by_transaction(
         &self,
+        cursor: EventID,
         digest: TransactionDigest,
         limit: usize,
+        reverse: bool,
     ) -> Result<Vec<StoredEvent>, SuiError>;
 
     /// Returns at most `limit` events of a certain EventType
     /// (e.g. `TransferObject`) within [start_time, end_time),
     /// sorted in in ascending time.
     async fn events_by_type(
         &self,
-        start_time: u64,
-        end_time: u64,
+        cursor: EventID,
         event_type: EventType,
         limit: usize,
+        reverse: bool,
     ) -> Result<Vec<StoredEvent>, SuiError>;
 
     /// Returns at most `limit` events emitted in a certain Module ID during
     /// [start_time, end_time), sorted in ascending time.
     async fn events_by_module_id(
         &self,
-        start_time: u64,
-        end_time: u64,
+        cursor: EventID,
         module: &ModuleId,
         limit: usize,
+        reverse: bool,
     ) -> Result<Vec<StoredEvent>, SuiError>;
 
     /// Returns at most `limit` events with the move event struct name
     /// (e.g. `0x2::devnet_nft::MintNFTEvent`) emitted
     /// during [start_time, end_time), sorted in ascending time.
     async fn events_by_move_event_struct_name(
         &self,
-        start_time: u64,
-        end_time: u64,
+        cursor: EventID,
         move_event_struct_name: &str,
         limit: usize,
+        reverse: bool,
     ) -> Result<Vec<StoredEvent>, SuiError>;
 
     /// Returns at most `limit` events associated with a certain sender
     /// emitted during [start_time, end_time), sorted in ascending time.
     async fn events_by_sender(
         &self,
-        start_time: u64,
-        end_time: u64,
+        cursor: EventID,
         sender: &SuiAddress,
         limit: usize,
+        reverse: bool,
```

Comment:
Can we name this parameter `ascending` instead of `reverse`?  I feel this is more obvious.  ascending true means return items in ascending sequence order, but `reverse` it is hard to tell what that means (ie what is normal?)

Reply:
Or rather maybe `descending`.....